internal/kafka/internal/workers/kafka_mgrs/provisioning_kafkas_mgr.go

```diff
@@ -56,10 +67,77 @@ func (k *ProvisioningKafkaManager) Reconcile() []error {
 	} else {
 		glog.Infof("provisioning kafkas count = %d", len(provisioningKafkas))
 	}
+
 	for _, kafka := range provisioningKafkas {
 		glog.V(10).Infof("provisioning kafka id = %s", kafka.ID)
+		if kafka.ClusterID == "" {
+			if err := k.reassignProvisioningKafka(kafka); err != nil {
+				encounteredErrors = append(encounteredErrors, errors.Wrapf(err, "failed to reconcile provisioning kafka %s", kafka.ID))
+				continue
+			}
+		}
 		metrics.UpdateKafkaRequestsStatusSinceCreatedMetric(constants2.KafkaRequestStatusProvisioning, kafka.ID, kafka.ClusterID, time.Since(kafka.CreatedAt))
 	}
 
 	return encounteredErrors
 }
+func (k *ProvisioningKafkaManager) reassignProvisioningKafka(kafka *dbapi.KafkaRequest) error {
+	cluster, e := k.clusterPlacementStrategy.FindCluster(kafka)
+	if e != nil || cluster == nil {
+		msg := fmt.Sprintf("No available cluster found for '%s' Kafka instance in region: '%s'", kafka.InstanceType, kafka.Region)
+		logger.Logger.Infof(msg)
+		return errors.New(fmt.Sprintf("Region %s cannot accept instance type: %s at this moment", kafka.Region, kafka.InstanceType))
+	}
+
+	kafka.ClusterID = cluster.ClusterID
+
+	truncatedKafkaIdentifier := fmt.Sprintf("%s-%s", services.TruncateString(kafka.Name, services.TruncatedNameLen), strings.ToLower(kafka.ID))
+
+	truncatedKafkaIdentifier, replaceErr := services.ReplaceHostSpecialChar(truncatedKafkaIdentifier)
+	if replaceErr != nil {
+		return errors.Wrapf(replaceErr, "generated host is not valid")
+
+	}
+
+	clusterDNS, err := k.clusterService.GetClusterDNS(kafka.ClusterID)
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving cluster DNS for kafka %s", kafka.ID)
+	}
+
+	clusterDNS = strings.Replace(clusterDNS, constants2.DefaultIngressDnsNamePrefix, constants2.ManagedKafkaIngressDnsNamePrefix, 1)
+
+	kafka.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, clusterDNS)
+
+	if k.kafkaConfig.EnableKafkaExternalCertificate {
+		// If we enable KafkaTLS, the bootstrapServerHost should use the external domain name rather than the cluster domain
+		kafka.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, k.kafkaConfig.KafkaDomainName)
+	}
+
+	latestStrimziVersion, latestStrimziVersionErr := cluster.GetLatestAvailableAndReadyStrimziVersion()
+	if latestStrimziVersionErr != nil {
+		return errors.Wrapf(latestStrimziVersionErr, "error finding ready strimzi versions for kafka %s", kafka.ID)
+	}
+	if latestStrimziVersion == nil {
+		return errors.Wrapf(err, "could not find latest strimzi version for kafka %s", kafka.ID)
```

Comment:
I believe `err` is nil here. It is last assigned here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/6f58345f7aaebd09dc13ccff48c410f97646c6a1/internal/kafka/internal/workers/kafka_mgrs/provisioning_kafkas_mgr.go#L102